### PR TITLE
Feat/pay 171/plaid web wallet

### PR DESCRIPTION
**Pull request by blockdylanb**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/services/deepLinkListener/contexts/DeelLinkListenerContext/index.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export * from './DeelLinkListenerContext.context'
 export type { DeepLinkListenerContext } from './DeelLinkListenerContext.types'
+export { DeepLinkClickState } from './DeelLinkListenerContext.types'
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/services/deepLinkListener/contexts/DeelLinkListenerContext/index.ts:3]: should it be 'DeelLink...' or 'DeepLink'

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/ThreeDSHandlerStripe/index.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { getData } from './selectors'
 import Success from './template.success'
 
 const ThreeDSHandlerStripe = ({ data, ...props }: Props) => {
+  // TODO pass the logic to here
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/modals/BuySell/ThreeDSHandlerStripe/index.tsx:15]: Still a TODO?

File: packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/AddBankYapily/SelectBank/index.tsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { WalletFiatType } from '@core/types'
 import DataError from 'components/DataError'
 import { actions, selectors } from 'data'
 import { RootState } from 'data/rootReducer'
+import { BankPartners } from 'data/types'
```
Comment (TheLeoB) [inline packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/AddBankYapily/SelectBank/index.tsx:9]: Is this used in this file?

File: packages/blockchain-wallet-v4-frontend/src/modals/BuySell/ThreeDSHandlerStripe/index.tsx (hunk 2 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/modals/BuySell/ThreeDSHandlerStripe/index.tsx]: yeah, but that code wasn't from me, development hadn't been updated with recently releases. I got all the branches up to date so this shouldn't be a code addition now.

File: packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/AddBankYapily/SelectBank/index.tsx (hunk 3 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/AddBankYapily/SelectBank/index.tsx]: fixed

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts [hunk 4]
```diff
@@ -72,40 +74,53 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
   }
 
   const conditionalRetry = function* (id: string) {
-    const { RETRY_AMOUNT, SECONDS } = POLLING
-    const data = yield retry(RETRY_AMOUNT, SECONDS * 1000, transferAccountState, id)
-    return data
-  }
+    const MAX_TRIES = Array(6)
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:77]: why isn't this a number?

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts [hunk 5]
```diff
@@ -72,40 +74,53 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
...
+    }
+  }
 
-      if (typeof account === 'string' && bankCredentials) {
+  const getBankUpdatePayload = function* (
+    bankCredentials: BankCredentialsType,
+    account: string | PlaidAccountType | YodleeAccountType
+  ) {
+    switch (true) {
+      case typeof account === 'string' && bankCredentials:
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:100]: what else can the `account` be?

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts [hunk 6]
```diff
@@ -72,40 +74,53 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
...
+        return [bankCredentials.id, account]
+    }
+  }
 
+  const fetchBankTransferUpdate = function* ({
+    payload
+  }: ReturnType<typeof A.fetchBankTransferUpdate>) {
+    try {
+      const bankCredentials = S.getBankCredentials(yield select()).getOrFail(
+        'Could not retrieve bank credentials'
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:121]: add this to a model somewhere, and make it an `enum`

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts [hunk 7]
```diff
@@ -143,31 +158,31 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
             account: bankData
           })
         )
+        const sbMethodsR = selectors.components.buySell.getBSPaymentMethods(yield select())
+        const sbMethods = sbMethodsR.getOrElse(DEFAULT_METHODS)
+        if (Remote.Success.is(sbMethodsR) && sbMethods.methods.length) {
+          const bankTransferMethod = sbMethods.methods.filter(
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:164]: why don't you use `find` here?

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts [hunk 8]
```diff
@@ -180,17 +195,53 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
     }
   }
 
+  const setupBankTransferProvider = function* () {
+    const fiatCurrency = S.getFiatCurrency(yield select()) || 'USD'
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:199]: this currency can be an `enum`, right? (maybe I'm just wrong)

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/selectors.ts [hunk 9]
```diff
@@ -28,14 +27,25 @@ export const getFiatCurrency = (state: RootState) => state.components.brokerage.
 
 export const getIsFlow = (state: RootState) => state.components.brokerage.isFlow
 
+export const getPlaidWalletHelperLink = createSelector(
+  (state: RootState) => selectors.core.walletOptions.getDomains(state),
+  (domainsR) => {
+    const { walletHelper } = domainsR.getOrElse({
+      walletHelper: 'https://wallet-helper.blockchain.com'
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/selectors.ts:34]: should be an enum? something to discuss

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/slice.ts [hunk 10]
```diff
@@ -42,16 +41,14 @@ const brokerageSlice = createSlice({
   reducers: {
     createFiatDeposit: () => {},
     deleteSavedBank: (state, action: PayloadAction<BankTransferAccountType['id']>) => {},
-    fetchBTUpdateLoading: (state) => {
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/slice.ts:45]: what is BTU?

Comment (dylanbathurst) [reply packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/slice.ts:45]: It's just "BT" bank transfer update

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts (hunk 4 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:77]: For my `conditionalRetry` function I didn't want to deal with decrementing and checking if the value was at 0 so i just used an array + array.pop() + a while loop to create the effect I wanted.

File: packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts (hunk 5 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/sagas.ts:100]: `account` is what wallet receives from the iframe's postMessage so it's different for each provider (plaid, yapily, yodlee). Now that i'm thinking about it though, if I just standardize that I would be able to get rid of some duplicate code. brb! 🏃🏻

**Review by TheLeoB — DISMISSED** [review]

**Pull request merged**